Add Columns selection element

danielajass opened the pull request:
- add an option to select columns in the table.

Commit:
Add Columns selection element

- add an option to select columns in the table.

mrshu commented:
@danielajass would you mind re-running `black` through these changes?

Commit:
Reformat the code with `black`

danielajass commented:
> @danielajass would you mind re-running `black` through these changes?

I did. Now it passed.

mrshu commented:
Thanks @danielajass!

The only question/suggestion I would have would be whether it would make sense to save the columns to show with the dataset, so that one wouldn't need to pick them at all times. Do you think it would make sense to add something like that?

samsucik committed:
make previous changes more consistent with existing code

danielajass commented:
> The only question/suggestion I would have would be whether it would make sense to save the columns to show with the dataset, so that one wouldn't need to pick them at all times. Do you think it would make sense to add something like that?

@mrshu, good idea! I looked at the code and looks doable. I'll add it.

Commit:
Enable save selected columns into datafile

- save selected columns into a datafile,
- read saved columns when a datafile is loaded.

Commit:
Fix naming

- rename the variable `COL_TO_SHOW` to `COLS_TO_SHOW`,
- use the variable instead of the hardcoded value.

samsucik commented on prompterator/utils.py:
```diff
@@ -186,6 +186,7 @@ def save_labelled_data():
             c.PROMPT_COMMENT_KEY: st.session_state[c.PROMPT_COMMENT_KEY],
             c.MODEL_KEY: model.name,
             c.PROMPT_CREATOR_KEY: get_prompt_creator(),
+            c.COL_TO_SHOW_KEY: st.session_state["columns_to_show"],
```
```suggestion
            c.COLS_TO_SHOW_KEY: st.session_state[c.COLS_TO_SHOW_KEY],
```

samsucik reviewed (approved):
All looks good; I haven't manually tested Prompterator with the latest version of these changes but I trust you on this as you're essentially making these changes to enable your own work 😄

danielajass committed:
Update prompterator/utils.py

Co-authored-by: Sam Sucik <ssucik@cisco.com>
